sw/host/opentitanlib/src/bootstrap/legacy.rs

```diff
@@ -277,7 +277,7 @@ impl UpdateProtocol for Legacy {
                 continue;
             }
 
-            if response.iter().all(|&x| x == response[0]) {
+            if response[..Frame::HASH_LEN].iter().all(|&x| x == response[0]) {
```

Comment:
If you're using an iterator, you can use `.take(..)` instead.

Reply:
Good to know, but I do not think there is a strong case for one being more clear than the other in this case.  Do you mind it gets merged in its current form.  (I can only test on someone else's Andreiboard, as mine does not use CRW signing.)